src/geocoder/geocoder.ts

```diff
@@ -741,7 +741,10 @@ module powerbi.extensibility.geocoder {
 
         export function getCoordinates(key: string): IGeocodeCoordinate {
             if (key) {
-                return ensureCache().getCoordinates(key);
+                let coordinates: IGeocodeCoordinate = undefined;
+                ensureCache().getCoordinates(key).then((data: IGeocodeCoordinate) => coordinates = data);
```

Comment:
Promise is supposed to be async. That means coordinates will be undefined.

Can we use await here?